src/include/uapi/ipc/dai-intel.h

```diff
@@ -129,6 +131,7 @@ struct sof_ipc_dai_hda_params {
  * data integrity problems.
  */
 struct sof_ipc_dai_dmic_pdm_ctrl {
+	struct sof_ipc_hdr hdr;
```

Comment:
@lgirdwood 
We don't need this one, it adds header for every pdm in `sof_ipc_dai_dmic_params`, but I don't know if it's good moment to again break abi, maybe we should get rid of it after release.
`sof_ipc_dai_dmic_pdm_ctrl` is just field in ipc message, maybe we should mark something like that so it won't look like standalone ipc command.

Reply:
We need it for size ti help with validation, I know it's repeated within the flex array but omitting it could cause confusion over "which structure needs size". Besides if the size does increase we can maintain some level of backwards compatibility by indexing the flex array using (void*)x + size * n rather than by x[n].